redbot/cogs/mod/kickban.py

```diff
@@ -260,29 +291,32 @@ async def ban(
         guild = ctx.guild
         if days is None:
             days = await self.config.guild(guild).default_days()
+        if isinstance(user, int):
+            try:
+                user = await self.bot.fetch_user(user)
+            except discord.errors.NotFound:
+                await ctx.send_help()
+                return
```

Comment:
This method should not try to fetch a user on ban as fetch user rate limit is low. It should (similarly to `[p]hackban`) use `discord.Object`s for banning when we only have a raw user ID.